### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by magnurud**
Latest and greatest changes

### CODE REVIEW

File: Lucidtech/Las/ApiClient.cs [hunk 1]
```diff
@@ -45,10 +44,13 @@ public class ApiClient : Client
...
         /// </returns>
-        public Prediction Predict(string documentPath, string modelName, string consentId)
+        public Prediction Predict(string documentPath, string modelName, string consentId = "default")
         {
             string contentType = GetContentType(documentPath);
-            string documentId = UploadDocument(documentPath, contentType, consentId);
+            byte[] body = File.ReadAllBytes(documentPath);
+            var postDocumentsResponse = JsonSerialPublisher.ObjectToDict<Dictionary<string, object>>(
+                PostDocuments(body, contentType, consentId));
+            string documentId = (string)postDocumentsResponse["documentId"];
```
Comment (krigar1184) [inline Lucidtech/Las/ApiClient.cs]: If `documentId` key is not present in the response, this line will throw an exception.

File: Lucidtech/Las/Core/Credentials.cs [hunk 2]
```diff
@@ -5,67 +5,162 @@
...
-    /// 2. AccessKeyId, SecretAccessKey and ApiKey are provided
+    /// 1. ClientId, ClientSecret, ApiKey, AuthEndpoint and ApiEndpoint are provided
+    /// 2. The path to the file where the credentials are stored is provided
     /// 3. Credentials are located in default path ~/.lucidtech/credentials.cfg
     /// 
     /// Get credentials by contacting hello@lucidtech.ai
     /// 
     /// </summary>
-    public class Credentials
+    public class AmazonCredentials
```
Comment (krigar1184) [inline Lucidtech/Las/Core/Credentials.cs]: Would be cool to have a possibility to override the credentials with environment variables, could you add this?

File: Lucidtech/Las/ApiClient.cs (hunk 1 above)
Comment (magnurud) [reply Lucidtech/Las/ApiClient.cs:53]: This seems to be the case in the python sdk as well, should we agree on a way to handle this consistently? @augustak

File: Lucidtech/Las/Core/Credentials.cs (hunk 2 above)
Comment (magnurud) [reply Lucidtech/Las/Core/Credentials.cs:25]: Should this be the default behaviour?

File: Lucidtech/Las/ApiClient.cs (hunk 1 above)
Comment (augustak) [reply Lucidtech/Las/ApiClient.cs:53]: documentId will always be present

File: Lucidtech/Las/Client.cs [hunk 3]
```diff
@@ -20,121 +20,153 @@ namespace Lucidtech.Las
...
+        /// <param name="feedback"> A list of feedback items {label: value},
+        /// representing the ground truth values for the document </param>
         /// <returns>
         /// A deserialized object that can be interpreted as a Dictionary with the fields
-        /// with documentId, uploadUrl, contentType and consentId
+        /// with batchId, documentId, contentType and consentId
         /// </returns>
-        ///         
-        public object PostDocuments(string contentType, string consentId)
+        public object PostDocuments(byte[] content, string contentType, string consentId,
```
[the remark this answers is not in the document]
Comment (magnurud) [reply Lucidtech/Las/Client.cs:63]: I suggest we use the same convention as the other SDKs, did you see any other examples of this?

File: Lucidtech/Las/ApiClient.cs [hunk 4]
```diff
@@ -45,10 +44,13 @@ public class ApiClient : Client
         /// <returns>
         /// Prediction on document
         /// </returns>
-        public Prediction Predict(string documentPath, string modelName, string consentId)
+        public Prediction Predict(string documentPath, string modelName, string consentId = "default")
```
[the remark this answers is not in the document]
Comment (magnurud) [reply Lucidtech/Las/ApiClient.cs:47]: My understanding is that the user should be able to consider this an optional variable. I believe that it is done in the same way for the other SDK's as well.

**Pull request merged**